### ISSUE TITLE
Catch typos like `1_32` instead of `1_i32`

### ISSUE THREAD

**Issue by scottmcm** [rust-lang-nursery/rust-clippy#3091]
I missed a letter in a very subtle way that still compiled, so it might be a good candidate for linting:
```rust
let mut x = 1_32;
```
when I meant to type
```rust
let mut x = 1_i32;
```
This, of course, made it do something _very_ different from what I wanted...

**Comment by mipli** [comment]
Just had an issue with this myself, so I wouldn't mind solving this one.

Been looking a bit into it, and thinking that it would fit to add it as another lint in the `clippy_lints/src/literal_representation.rs` file. 

What I'm wondering though is what the name of the new lint should be. `misstyped_literal`, `misstyped_literal_suffix`?

And I'm assuming it should go into the `correctness` group, due to the label on this issue.

**Comment by flip1995** [comment]
>Been looking a bit into it, and thinking that it would fit to add it as another lint in the clippy_lints/src/literal_representation.rs file.

Yes sounds good!

I think `misstyped_literal_suffix` is a good name for this. 

`correctness` feels like the right lint group for this.

**Comment by oli-obk** [comment]
Naming conventions say it should be `mistyped_literal_suffixes` so `allow(mistyped_literal_suffixes)` sounds like an english sentence

**Comment by dtolnay** [comment]
Added in #3129.
